apps/dapp-console-api/src/middleware/isGithubAuthed.ts

```diff
@@ -0,0 +1,64 @@
+import { TRPCError } from '@trpc/server'
+
+import type { GrowthbookStore } from '@/growthbook'
+import { metrics } from '@/monitoring/metrics'
+import { Trpc } from '@/Trpc'
+
+/** Middleware used for checking if the request is from a privy user with a github account linked. */
+export const isGithubAuthed = (
+  trpc: Trpc,
+  growthBookStore: GrowthbookStore,
+) => {
+  return trpc.middleware(async ({ ctx, next }) => {
+    const { session } = ctx
+
+    if (!growthBookStore.get('enable_github_auth')) {
+      return next()
+    }
+
+    if (!session.user || !session.user.privyDid) {
+      throw new TRPCError({
+        code: 'UNAUTHORIZED',
+        message: 'User is not logged into privy',
+      })
+    }
+
+    if (!!session.user.githubSubject) {
```

Comment:
what does this do?

Reply:
if the github subject is already stored on the user's session it bypasses making a fetch to privy for the user's github info

Reply:
why does this call `next()`? It seems like below it's updated